File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        
        int maximumSum = nums[0];
        int tempSum = 0;
        
        for(int i=0; i<nums.size(); i++){
            if(tempSum<0) tempSum = 0;
            tempSum += nums[i];
            maximumSum = std::max(maximumSum, tempSum);
        }
        
        return maximumSum;
    }
};

File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maximum_sum: int = min(nums)
        _sum: int = 0
        
        for num in nums:
            if _sum < 0:
                _sum = 0
            _sum += num
            maximum_sum = max(_sum, maximum_sum)
        
        return maximum_sum
        
        